kuka_rsi_hw_interface/src/kuka_hardware_interface_node.cpp

```diff
@@ -54,62 +54,43 @@ int main(int argc, char** argv)
   kuka_rsi_hw_interface.configure();
 
   // Set up timers
-  struct timespec ts = {0, 0};
-  if (clock_gettime(CLOCK_MONOTONIC, &ts) != 0)
-  {
-    ROS_FATAL("Failed to poll clock!");
-  }
-  ros::Time last(ts.tv_sec, ts.tv_nsec);
-  ros::Time now(ts.tv_sec, ts.tv_nsec);
-  ros::Duration period(1.0);
+  ros::Time ros_now;
+  ros::Duration ros_period;
+  auto rt_last = std::chrono::steady_clock::now();
+  auto rt_now = rt_last;
 
   controller_manager::ControllerManager controller_manager(&kuka_rsi_hw_interface, nh);
 
   kuka_rsi_hw_interface.start();
 
   // Get current time and elapsed time since last read
-  if (!clock_gettime(CLOCK_MONOTONIC, &ts))
-  {
-    now.sec = ts.tv_sec;
-    now.nsec = ts.tv_nsec;
-    period = now - last;
-    last = now;
-  }
-  else
-  {
-    ROS_FATAL("Failed to poll clock!");
-  }
+  ros_now = ros::Time::now();
+  rt_now = std::chrono::steady_clock::now();
+  ros_period.fromSec(std::chrono::duration_cast<std::chrono::duration<double>>(rt_now - rt_last).count());
```

Comment:
I prefixed the period with ros because it's a `ros::Duration` object. It's my own way of keeping track of `std::chrono` and `ros` types for timekeeping. I'm definitely open to changing the naming.